tinygrad/tensor.py

```diff
@@ -225,6 +226,9 @@ def chunk(self, num, dim):
   # TODO: what's the difference between dot and matmul?
   def dot(self:Tensor, w:Tensor):
     # NOTE: we use a 1x1 conv2d to do the matmul. mxk @ kxn = (1,k,m,1).conv2d(n,k,1,1)
+    assert self.shape != (0,) and w.shape != (0,), f"Shape of tensors must not be {self.shape}"
+    assert self.shape == w.shape, f"Shape of tensors must match"
```

Comment:
Huh, this isn't true.

Reply:
oops, sorry I did that to quickly lol. Let me fix it.